lxd/storage/backend_lxd.go

```diff
@@ -3723,25 +3726,118 @@ func (b *lxdBackend) ImportBucket(projectName string, poolVol *backupConfig.Conf
 	}
 
 	// Validate config and create database entry for restored bucket.
-	bucketID, err := BucketDBCreate(context.TODO(), b, projectName, true, bucket)
+	bucketID, err := BucketDBCreate(b.state.ShutdownCtx, b, projectName, true, bucket)
 	if err != nil {
 		return nil, err
 	}
 
-	revert.Add(func() { _ = BucketDBDelete(context.TODO(), b, bucketID) })
+	revert.Add(func() { _ = BucketDBDelete(b.state.ShutdownCtx, b, bucketID) })
 
 	// Get the bucket name on storage.
-	storageBucketName := project.StorageVolume(projectName, poolVol.Bucket.Name)
+	storageBucketName := project.StorageVolume(projectName, bucket.Name)
 	storageBucket := b.GetVolume(drivers.VolumeTypeBucket, drivers.ContentTypeFS, storageBucketName, bucketConfig)
 
 	err = b.driver.ValidateVolume(storageBucket, false)
 	if err != nil {
 		return nil, err
 	}
 
+	// Recover bucket keys.
+
+	// Start minio process.
+	minioProc, err := b.ActivateBucket(projectName, bucket.Name, op)
+	if err != nil {
+		return nil, err
+	}
+
+	// Initialize minio client object.
+	adminClient, err := minioProc.AdminClient()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, ctxCancel := context.WithTimeout(b.state.ShutdownCtx, time.Duration(time.Second*30))
+	defer ctxCancel()
+
+	// Export IAM data (response is ZIP file).
```

Comment:
Can this section below be moved into its own function on backendLXD called something like `recoverMinIOKeys`